src/btree/row_srch.c

```diff
@@ -116,6 +127,40 @@ __wt_search_insert(
             last_ins = ins;
             key.data = WT_INSERT_KEY(ins);
             key.size = WT_INSERT_KEY_SIZE(ins);
+            /*
+             * We have an optimization to reduce the number of bytes we need to compare down the
+             * search if we know certain length of the prefix of the search key matches the keys we
+             * have already compared upper the stack. This works because we know the keys become
```

Comment:
I suggest a little rephrasing here:  "We have an optimization to reduce the number of bytes we need to compare **during a** search if we know **a prefix** of the search key matches the keys we have already compared **on the upper stacks.** "